### ISSUE TITLE
bug: Lack of compatibility between variable `umi_federated_credentials_advanced` and `federated_credentials_advanced` for `audience` sub-variable

### ISSUE THREAD

**Issue by tcharewicz** [Azure/terraform-azurerm-lz-vending#299]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Versions

Please paste the output of `terraform version` command from within the initialized directory:

```text
Terraform v1.5.5
on linux_arm64
+ provider registry.terraform.io/azure/azapi v1.10.0
+ provider registry.terraform.io/hashicorp/azuread v2.46.0
+ provider registry.terraform.io/hashicorp/azurerm v3.84.0
```

Please enter the module version that you are using:

```text
3.4.2
```

### Description

I would like to configure UMI with Federated Credentials (Advanced), and I think it is not possible to do while to use main module. It require to configure variable `umi_federated_credentials_advanced` with defined type where sub-variable `audience` is a string. This variable is passed directly to sub-module in this repository where variable `federated_credentials_advanced` has a different type, there is no sub-variable `audience` but there is a sub-variable `audiences` with type set of strings. 

- variable in main module
```
variable "umi_federated_credentials_advanced" {
  type = map(object({
    name               = string
    subject_identifier = string
    issuer_url         = string
    audience           = optional(string, "api://AzureADTokenExchange") <--- string
  }))
  default     = {}
  description = <<DESCRIPTION
Configure federated identity credentials, using OpenID Connect, for use scenarios outside GitHub Actions and Terraform Cloud.

The may key is arbitrary and only used for the `for_each` in the resource declaration.

The map value is an object with the following attributes:

- `name`: The name of the federated credential resource, the last segment of the Azure resource id.
- `subject_identifier`: The subject of the token.
- `issuer_url`: The URL of the token issuer, should begin with `https://`
- `audience`: (optional) The token audience, defaults to `api://AzureADTokenExchange`.
DESCRIPTION
}
```

- variable in sub-module where UMI is created
```
variable "federated_credentials_advanced" {
  type = map(object({
    name               = string
    subject_identifier = string
    audiences          = optional(set(string), ["api://AzureADTokenExchange"]) <--- list of strings
    issuer_url         = string
  }))
  default     = {}
  description = <<DESCRIPTION
Configure federated identity credentials, using OpenID Connect, for use scenarios outside GitHub Actions and Terraform Cloud.

The may key is arbitrary and only used for the `for_each` in the resource declaration.

The map value is an object with the following attributes:

- `name` - the name of the federated credential resource, the last segment of the Azure resource id.
- `subject_identifier` - The subject of the token.
- `issuer_url` - the URL of the token issuer, should begin with `https://`
- `audiences` - (optional) a set of strings containing the token audiences, defaults to `["api://AzureADTokenExchange"]`.
DESCRIPTION
}
```

#### Steps to Reproduce

1. Check code in repository

#### Additional context

It is possible to create UMI with federated credentials (Advanced) while using directly sub-module for usermanagedidentity.

**Comment by tcharewicz** [comment]
Pull Request https://github.com/Azure/terraform-azurerm-lz-vending/pull/300

**Comment by matt-FFFFFF** [comment]
Closing as merged #300
